src/blockdata/locktime.rs

```diff
@@ -134,30 +135,39 @@ impl From<PackedLockTime> for u32 {
     }
 }
 
-impl FromStr for PackedLockTime {
-    type Err = ParseIntError;
+/// Implements `TryFrom<$from> for $to` using `parse_int`, mapping the output using `fn`
+macro_rules! impl_tryfrom_str_single {
+    ($($from:ty, $to:ident $(, $fn:ident)?);*) => {
+        $(
+        impl TryFrom<$from> for $to {
+            type Error = ParseIntError;
```

Comment:
If we change this to `type Error = Error` and add an extra `?` to line 146 below
```rust
    parse_int(s).map($to $(:: $fn)?)?
```
We can then use `impl_tryfrom_str!` for `Height` and `Time` as well. But ... we need the slightly ugly helper functions
```rust
     // Helper function used by `impl_tryfrom_str!`.
     fn _from_consensus(n: u32) -> Result<LockTime, Error> {
         Ok(Self::from_consensus(n))
     }
```
and
```rust
     // Helper function used by `impl_tryfrom_str!`.
     fn _from_u32(n: u32) -> Result<PackedLockTime, Error> {
         Ok(Self::from_consensus(n))
     }
```
so we have
```rust
impl_tryfrom_str!(PackedLockTime, _from_u32);
impl_tryfrom_str!(LockTime, _from_consensus);
```

Reply:
Was thinking about de-duplicating but couldn't figure out anything that's not horrible. Maybe with proc macros it'd be fine but those are slooooow to compile.